src/libstd/sys/wasi/os.rs

```diff
@@ -12,36 +12,20 @@ use crate::sys::memchr;
 use crate::sys::{cvt, unsupported, Void};
 use crate::vec;
 
+use wasi::wasi_unstable as wasi;
+
 #[cfg(not(target_feature = "atomics"))]
 pub unsafe fn env_lock() -> impl Any {
     // No need for a lock if we're single-threaded, but this function will need
     // to get implemented for multi-threaded scenarios
 }
 
 pub fn errno() -> i32 {
-    extern {
-        #[thread_local]
-        static errno: libc::c_int;
-    }
-
-    unsafe { errno as i32 }
+    0
 }
 
 pub fn error_string(errno: i32) -> String {
-    extern {
-        fn strerror_r(errnum: libc::c_int, buf: *mut libc::c_char,
```

Comment:
I think it's best to call into `libc` for this, I'm not sure why we'd want to duplicate what's already there?

Reply:
I think human-readable error descriptions are a good thing to have in `wasi` either way, so why not use it? Plus we will not have to jump through all those C-hoops.